releases/oidc-ingress.yaml

```diff
@@ -0,0 +1,292 @@
+- name: "kubernetes-incubator"
+  url: "https://kubernetes-charts-incubator.storage.googleapis.com"
+- name: "forecastle"
+  # Cannot use release tags, see https://github.com/aslafy-z/helm-git/issues/9
+  # url: "git+https://github.com/stakater/Forecastle@deployments/kubernetes/chart?ref=v1.0.25"
+  url: "git+https://github.com/stakater/Forecastle@deployments/kubernetes/chart?ref=8f36b82beaf2a1a42b364a3857bc83638c51e30b"
+
+releases:
+
+################################################################################
+## oidc-ingress                                                               ##
+## oidc-ingress is a package of resources that implement an ingress protected ##
+## by an OIDC authorization server that authenticates users and distributes   ##
+## authorizations in the form of OIDC claims (by default, "roles").           ##
+################################################################################
+
+#############################################################################
+## Security consideration
+##
+## Ensure your use of oidc-ingress and the configuration of your clients and
+## authorization server guard against the attacks described in RFC 6749.
+##  https://tools.ietf.org/html/rfc6749#section-10.6
+##
+##############################################################################
+
+
+#######################################################################################
+## keycloak-gatekeeper (formerly keycloak-proxy)                                     ##
+## keycloak-gatekeeper is an open source identity-aware proxy                        ##
+#######################################################################################
+
+#
+# References:
+#   - https://github.com/gabibbo97/charts/tree/master/charts/keycloak-gatekeeper
+#   - https://hub.helm.sh/charts/gabibbo97/keycloak-gatekeeper
+#   - https://www.keycloak.org/
+#
+- name: "oidc-ingress"
+  namespace: "oidc-ingress"
+  labels:
+    chart: "keycloak-gatekeeper"
+    repo: "gabibbo97"
+    component: "oidc-ingress"
+    namespace: "oidc-ingress"
+    vendor: "keycloak"
+    default: "true"
+  chart: "gabibbo97/keycloak-gatekeeper"
+  version: "1.2.1"
+  wait: false
+  installed: {{ env "OIDC_INGRESS_INSTALLED" | default "true" }}
+  values:
+  - nameOverride: "oidc-ingress"
+    fullNameOverride: "oidc-ingress"
+    image:
+      tag: 6.0.1
+      pullPolicy: "IfNotPresent"
+    debug: {{ env "OIDC_INGRESS_DEBUG" | default "false" }}
+    replicas: {{ env "OIDC_INGRESS_REPLICAS" | default 1 }}
+    logging: false
+    sessionCookies: false
+    droolsPolicyEnabled: false
+    # You must set OIDC_INGRESS_DENY_ACCESS_BY_DEFAULT to "false" if you want
+    # to permit access to URLs not otherwise restricted.
+    defaultDeny: {{ env "OIDC_INGRESS_DENY_ACCESS_BY_DEFAULT" | default "true" }}
+    ingress:
+      enabled: true
+      annotations:
+        kubernetes.io/ingress.class: nginx
+        nginx.ingress.kubernetes.io/proxy-buffer-size: "16k"
```

Comment:
why was this value chosen?

Reply:
Default value was too small. Without doing extensive investigation/testing, this seems like a reasonable value, large enough to cover our expected uses, small enough to not take excessive resources.

Reply:
what happened when it was too small? just so I know what this toggle addresses

Reply:
This is copied from https://github.com/cloudposse/helmfiles/blob/82f1bcfbd0d6a05d6e0cc39c8cb98ded1c460e01/releases/keycloak-gatekeeper.yaml#L64

Default is, I think, 4k. The headers can get huge, particularly because the OIDC cookie can get huge, and when the request header exceeds the buffer size, `nginx` sends it in fragments. Some recipients do not properly handle fragmented request headers, leading to valid requests being rejected.

Reply:
aha! ok, that's helpful to know.